test/app-tap/datetime.test.lua

```diff
@@ -1127,6 +1317,135 @@ test:test("Matrix of allowed time and interval subtractions", function(test)
+        local year, month, day, str, date_part_len
+        year, month, day, str, date_part_len = unpack(value)
+        local expected_date = date.new{year = year, month = month, day = day}
+        local date_part, len
+        date_part, len = date.parse_date(str)
+        test:is(len, date_part_len, ('%s: length check %d'):format(str, len))
+        test:is(expected_date, date_part, ('%s: expected date'):format(str))
+    end
+end)
+
+test:test("Parse iso8601 date - invalid strings", function(test)
+    test:plan(29)
+    local bad = {
+        "20121232"   , -- Invalid day of month
+        "2012-12-310", -- Invalid day of month
+        "2012-13-24" , -- Invalid month
+        "2012367"    , -- Invalid day of year
+        "2012-000"   , -- Invalid day of year
+        "2012W533"   , -- Invalid week of year
+        "2012-W52-8" , -- Invalid day of week
+        "2012Q495"   , -- Invalid day of quarter
+        "2012-Q5-85" , -- Invalid quarter
+        "20123670"   , -- Trailing digit
+        "201212320"  , -- Trailing digit
+        "2012-12"    , -- Reduced accuracy
+        "2012-Q4"    , -- Reduced accuracy
+        "2012-Q42"   , -- Invalid
+        "2012-Q1-1"  , -- Invalid day of quarter
+        "2012Q--420" , -- Invalid
+        "2012-Q-420" , -- Invalid
+        "2012Q11"    , -- Incomplete
+        "2012Q1234"  , -- Trailing digit
+        "2012W12"    , -- Incomplete
+        "2012W1234"  , -- Trailing digit
+        "2012W-123"  , -- Invalid
+        "2012-W12"   , -- Incomplete
+        "2012-W12-12", -- Trailing digit
+        "2012U1234"  , -- Invalid
+        "2012-1234"  , -- Invalid
+        "2012-X1234" , -- Invalid
+        "-5879611-01-01",  -- Year less than 5879610-06-22
+        "5879612-01-01",  -- Year greater than 5879611-07-11
+    }
+
+    for _, str in ipairs(bad) do
+        assert_raises(test, invalid_date_fmt_error(str),
+                      function() date.parse_date(str) end)
+    end
+end)
+
+test:test("Parse tiny date into seconds and other parts", function(test)
+    test:plan(4)
+    local str = '19700101 00:00:30.528'
+    local tiny = date.parse(str)
+    test:is(tiny.epoch, 30, ("epoch of '%s'"):format(str))
+    test:is(tiny.nsec, 528000000, ("nsec of '%s'"):format(str))
+    test:is(tiny.sec, 30, "sec")
+    test:is(tiny.timestamp, 30.528, "timestamp")
+end)
+
+test:test("Parse strptime format", function(test)
+    test:plan(16)
+    local formats = {
+        {'Thu Jan  1 03:00:00 1970',    '%c',       '1970-01-01T03:00:00Z'},
+        {'01/01/70',                    '%D',       '1970-01-01T00:00:00Z'},
+        {'01/01/70',                    '%m/%d/%y', '1970-01-01T00:00:00Z' },
+        {'Thu Jan  1 03:00:00 1970',    '%Ec',      '1970-01-01T03:00:00Z' },
+        {'1970-01-01',                  '%F',       '1970-01-01T00:00:00Z' },
+        {'1970-01-01',                  '%Y-%m-%d', '1970-01-01T00:00:00Z' },
+        -- {' 1-Jan-1970',                 '%v' },
```

Comment:
Why it is commented?

Reply:
Good question - it used to not work at some stage before, but now it's working perfectly well

```
tarantool> T = date.new{year = 2000, month = 3, day = 20}
---
...

tarantool> T:format('%v')
---
- 20-Mar-2000
...

tarantool> date.parse(T:format('%v'), {format='%v'})
---
- 2000-03-20T00:00:00Z
- 11
...
```

Will restore it.

Reply:
Restored